src/device/si/si_controller.c

```diff
@@ -78,20 +69,9 @@ static void dma_si_read(struct si_controller* si)
 
     update_pif_read(si);
 
-    for (i = 0; i < PIF_RAM_SIZE; i += 4)
-    {
-        si->ri->rdram.dram[(si->regs[SI_DRAM_ADDR_REG]+i)/4] = sl(*(uint32_t*)(&si->pif.ram[i]));
-    }
-
     cp0_update_count(si->r4300);
-
-    if (si->delay_si) {
-        si->regs[SI_STATUS_REG] |= SI_STATUS_DMA_BUSY;
-        add_interrupt_event(&si->r4300->cp0, SI_INT, /*0x100*/0x900);
-    } else {
-        si->regs[SI_STATUS_REG] |= SI_STATUS_INTERRUPT;
-        signal_rcp_interrupt(si->r4300, MI_INTR_SI);
-    }
+    si->regs[SI_STATUS_REG] |= SI_STATUS_RD_BUSY;
```

Comment:
Not sure, but shouldn't SI_STATUS_DMA_BUSY be also set ?

Reply:
No, it doesn't work if you do that. SI_STATUS_RD_BUSY is supposed to be set during the PIF processing, and DMA_BUSY should be set when the DMA is happening. So what we are emulating with this PR, is during a read, we set RD_BUSY until the interrupt, and the emulate an "instant" DMA portion (we never set DMA_BUSY). We do the opposite for the write.

If we were trying to be a little better, we might have it flip from RD_BUSY->DMA_BUSY halfway through the timer, but doing it this way seems to work well.